core/src/main/java/org/apache/gravitino/authorization/OwnerManager.java

```diff
@@ -0,0 +1,193 @@
+                    }));
+      } else if (ownerType == Owner.Type.GROUP) {
+        NameIdentifier ownerIdent = AuthorizationUtils.ofGroup(metalake, ownerName);
+        TreeLockUtils.doWithTreeLock(
+            objectIdent,
+            LockType.READ,
+            () ->
+                TreeLockUtils.doWithTreeLock(
+                    ownerIdent,
+                    LockType.READ,
+                    () -> {
+                      store
+                          .relationOperations()
+                          .insertRelation(
+                              SupportsRelationOperations.Type.OWNER_REL,
+                              objectIdent,
+                              MetadataObjectUtil.toEntityType(metadataObject),
+                              ownerIdent,
+                              Entity.EntityType.GROUP,
+                              true);
+                      return null;
+                    }));
+      }
+    } catch (NoSuchEntityException nse) {
+      LOG.warn(
+          "Metadata object {} or owner {} is not found", metadataObject.fullName(), ownerName, nse);
+      throw new NotFoundException(
+          nse, "Metadata object %s or owner %s is not found", metadataObject.fullName(), ownerName);
+    } catch (IOException ioe) {
+      LOG.info(
+          "Fail to set the owner {} of metadata object {}",
+          ownerName,
+          metadataObject.fullName(),
+          ioe);
+      throw new RuntimeException(ioe);
+    }
+  }
+
+  public Optional<Owner> getOwner(String metalake, MetadataObject metadataObject) {
+    try {
+      OwnerImpl owner = new OwnerImpl();
+      NameIdentifier ident = MetadataObjectUtil.toEntityIdent(metalake, metadataObject);
+      List entities =
+          TreeLockUtils.doWithTreeLock(
+              ident,
+              LockType.READ,
+              () ->
+                  store
+                      .relationOperations()
+                      .listEntitiesByRelation(
+                          SupportsRelationOperations.Type.OWNER_REL,
+                          ident,
+                          MetadataObjectUtil.toEntityType(metadataObject)));
+
+      if (entities.isEmpty()) {
+        return Optional.empty();
+      }
+
+      if (entities.size() != 1) {
+        throw new IllegalStateException(
+            String.format("The number of the owner %s must be 1", metadataObject.fullName()));
+      }
+
+      if (entities.get(0) instanceof UserEntity) {
+        UserEntity user = (UserEntity) entities.get(0);
+        owner.name = user.name();
+        owner.type = Owner.Type.USER;
+      } else if (entities.get(0) instanceof GroupEntity) {
+        GroupEntity group = (GroupEntity) entities.get(0);
+        owner.name = group.name();
+        owner.type = Owner.Type.GROUP;
+      } else {
+        throw new IllegalArgumentException(
+            String.format(
+                "Doesn't support owner entity class %s", entities.get(0).getClass().getName()));
+      }
```

Comment:
What is the meaning of `else` here, seems like this `else` will catch everything and throw an exception, how will the code reach to line 168?

Reply:
If the entity is UserEntity or GroupEntity, the code won't reach the `else` code block.